pkg/git/git.go

```diff
@@ -110,79 +110,63 @@ func (cloneOpts *CloneOptions) Validate() error {
 	return nil
 }
 
-// Clone clones a git repository
-func (cloneOpts *CloneOptions) cloneBare() (*git.Repository, error) {
-	args := []string{"clone", "--bare"}
-	if cloneOpts.Commit != "" {
-		args = append(args, "--no-checkout")
+// CheckoutFromBare checks out the specified reference from a bare repository
+func (cloneOpts *CloneOptions) CheckoutFromBare() error {
```

Comment:
Hi @Manoramsharma 😃 

Good Job! I think there is no problem with the overall direction of this PR, and there are a few small details I think may need to change. First set this method `CheckoutFromBare()` to private, this package only exposes the `CloneOption` struct and `Clone` method, two public things.

Clone() method will do follow things:

1. Clone() a remote repo as a bare repo into a local path. `branch`, `commit` and `tag` will be ignored.
2. Clone() a remote repo as a normal repo and checkout to `branch`, `commit` and `tag`.
3. Clone() a bare repo as a bare repo into a local path. `branch`, `commit` and `tag` will be ignored.
4. Clone() a bare repo as a normal repo and checkout to `branch`, `commit` and `tag`.

And these are four test cases you need to add for the next comit.

Reply:
That's great to hear from you . Thanks @zong-zhe for the review!!

Excited to hear a positive response I was working in the same direction as the changes you suggested are already in my mind, will surely come up with required code changes in the next PR. 🙌🏻